crates/vfs-notify/src/lib.rs

```diff
@@ -258,3 +263,23 @@ fn read(path: &AbsPath) -> Option<Vec<u8>> {
 fn log_notify_error<T>(res: notify::Result<T>) -> Option<T> {
     res.map_err(|err| tracing::warn!("notify error: {}", err)).ok()
 }
+
+/// Is `path` a symlink to a parent directory?
+///
+/// Including this path is guaranteed to cause an infinite loop. This
+/// heuristic is not sufficient to catch all symlink cycles (it's
+/// possible to construct cycle using two or more symlinks), but it
+/// catches common cases.
+fn path_is_parent_symlink(path: &Path) -> bool {
+    // We can't canonicalize this path, presume it's not a symlink.
+    let Ok(canonical) = path.canonicalize() else {
+        return false;
+    };
+
+    // The canonicalized path is the same, it's not a symlink.
+    if path == canonical {
```

Comment:
This will always fail on windows, as canonicalizing paths on windows will give back an [extended length path](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file)

Reply:
Updated to use `read_link()` instead :)